pinot-common/src/main/java/org/apache/pinot/common/datablock/MetadataBlock.java

```diff
@@ -69,6 +71,17 @@ public static MetadataBlock newEosWithStats(List<DataBuffer> statsByStage) {
   public MetadataBlock(List<DataBuffer> statsByStage) {
```

Comment:
Do we want to remove this constructor eventually? Is there scenario where `stageId` and `workerId` are unavailable?